File: src/main/java/com/appointmed/dto/AppointmentRequest.java
package com.appointmed.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AppointmentRequest {
    @NotNull private Long patientId;
    @NotNull private Long doctorId;
    @NotNull private LocalDate appointmentDate;
    @NotNull private LocalTime appointmentTime;
    private LocalTime endTime;
    private String reason;
    private BigDecimal fee;
}

File: mobile/app/src/main/java/com/appointmed/mobile/data/model/AppointmentRequest.kt
package com.appointmed.mobile.data.model

data class AppointmentRequest(
    val patientId: Long,
    val doctorId: Long,
    val appointmentDate: String,  // yyyy-MM-dd
    val appointmentTime: String,  // HH:mm
    val endTime: String? = null,
    val reason: String? = null,
    val fee: Double? = null
)
